### PR TITLE
message size and blended tests

### PR DESCRIPTION
**Pull request by laynemoseley**

### CODE REVIEW

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+# Performance Experiment - MQ - Actor Count and Message Size Blend
+
+The goal of this experiment is to benchmark how this technology performs when hardware resources are plentiful and the number of actors (publishers and subscribers) in the system varies.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:3]:
```suggestion
The goal of this experiment is to benchmark how this technology performs when hardware resources are plentiful, the number of actors (publishers and subscribers) in the system varies, and the size of the message varies.
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+...
+```
+
+Link to your program’s usage here:
+
+[Replace me](Replace me)
+
+### Data Processor
+
+The experiment designers will provide a program that accepts a list of producer csv files and a list of consumer csv files. It should output 2 CSV’s that marry up the matching ID’s of messages and contains calculated results.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:61]:
```suggestion
The experiment designers will provide a program that accepts a list of producer csv files and a list of consumer csv files. It should output 2 CSV’s that marry up the matching ID’s of messages and contain calculated results.
```

fix my typo

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+If possible, implement these configurations with redundancy enabled and document the configuration.
+
+These tests are described in detail below and have placeholders for results.
+
+## Redundancy implementation Details
+
+Place the redundancy implementation details here.
+
+## 1 Producer - 1 Consumer - tiny message size
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:115]:
```suggestion
## 1 Producer - 1 Consumer - Tiny Message Size
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+Run one instance of your producer program on one VM.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:117]:
```suggestion
Run one instance of your producer program on one VM. Configure your producer to always send the [tiny](./messages/tiny.txt) message.
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+| Processing Throughput Min | |
+| Processing Throughput Max | |
+| Processing Throughput Average | |
+| Processing Throughput Standard Deviation | |
+| Send Throughput Min | |
+| Send Throughput Max | |
+| Send Throughput Average | |
+| Send Throughput Standard Deviation | |
+
+## 1 Producer VM - 1 Consumer VM: 5 Instances - small message size
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:153]:
```suggestion
## 1 Producer VM - 1 Consumer VM: 5 Instances - Small Message Size
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+Run one instance of your producer program on one VM.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:155]:
```suggestion
Run one instance of your producer program on one VM. Configure your producer to always send the [small](./messages/small.txt) message.
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+| Processing Throughput Min | |
+| Processing Throughput Max | |
+| Processing Throughput Average | |
+| Processing Throughput Standard Deviation | |
+| Send Throughput Min | |
+| Send Throughput Max | |
+| Send Throughput Average | |
+| Send Throughput Standard Deviation | |
+
+## 1 Producer VM: 5 Instances - 1 Consumer VM: 1 Instance - medium message size
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:191]:
```suggestion
## 1 Producer VM: 5 Instances - 1 Consumer VM: 1 Instance - Medium Message Size
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+
+Run five producer program instances on one VM.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:193]:
```suggestion
Run five producer program instances on one VM. Configure the producer to always send the [medium](./messages/medium.txt) message.
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+
+Ensure you run your programs in such a way that the output is not lost. For example: `python main.py > results.csv`.
+
+Run one instance of the producer program on one VM.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:197]:
```suggestion
Run one instance of the consumer program on one VM.
```
Fix my mistake

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+| Processing Throughput Min | |
+| Processing Throughput Max | |
+| Processing Throughput Average | |
+| Processing Throughput Standard Deviation | |
+| Send Throughput Min | |
+| Send Throughput Max | |
+| Send Throughput Average | |
+| Send Throughput Standard Deviation | |
+
+## 3 Producer VM's: 8 Instances - 3 Consumers VM's: 8 Instances - large message size
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:229]:
```suggestion
## 3 Producer VM's: 8 Instances - 3 Consumers VM's: 8 Instances - Large Message Size
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+
+Run eight instances of your producer on three different machines (24 instances total).
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:231]:
```suggestion
Run eight instances of your producer on three different machines (24 instances total). Configure your producers to always send the [large](./messages/large.txt) message.
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+| Processing Throughput Min | |
+| Processing Throughput Max | |
+| Processing Throughput Average | |
+| Processing Throughput Standard Deviation | |
+| Send Throughput Min | |
+| Send Throughput Max | |
+| Send Throughput Average | |
+| Send Throughput Standard Deviation | |
+
+## 4 Producer VM's: 25 Threads - 4 Consumer VM's: 25 Threads - x-large message size
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:271]:
```suggestion
## 4 Producer VM's: 25 Threads - 4 Consumer VM's: 25 Threads - X-Large Message Size
```

File: DesignOfExperiments/Performance-MQ-Blended.md [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+
+Run 25 instances of the producer program on four different VM's (100 instances total).
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-Blended.md:273]:
```suggestion
Run 25 instances of the producer program on four different VM's (100 instances total). Configure your producers to always send the [x-large](./messages/x-large.txt) message.
```

File: DesignOfExperiments/Performance-MQ-MessageSize.md [hunk 14]
```diff
@@ -0,0 +1,151 @@
+# Performance Experiment - MQ - Message Size
+
+The goal of this experiment is to benchmark how this technology performs when hardware resources are plentiful and the number of actors (publishers and subscribers) in the system varies.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-MessageSize.md:3]:
```suggestion
The goal of this experiment is to benchmark how this technology performs when hardware resources are plentiful and the size of messages varies.
```

File: DesignOfExperiments/Performance-MQ-MessageSize.md [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+...
+```
+
+Link to your program’s usage here:
+
+[Replace me](Replace me)
+
+### Data Processor
+
+The experiment designers will provide a program that accepts a list of producer csv files and a list of consumer csv files. It should output 2 CSV’s that marry up the matching ID’s of messages and contains calculated results.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-MessageSize.md:65]:
```suggestion
The experiment designers will provide a program that accepts a list of producer csv files and a list of consumer csv files. It should output 2 CSV’s that marry up the matching ID’s of messages and contain calculated results.
```
Fix my typo

File: DesignOfExperiments/Performance-MQ-MessageSize.md [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+| 1 | 1 | 1 | 1 | large |
+| 1 | 1 | 1 | 1 | x-large |
+
+If possible, implement these configurations with redundancy enabled and document the configuration.
+
+## Redundancy implementation Details
+
+Place the redundancy implementation details here.
+
+## 1 Producer - 1 Consumer
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-MessageSize.md:115]:
```suggestion
## 1 Producer - 1 Consumer - Tiny Message Size
```

File: DesignOfExperiments/Performance-MQ-MessageSize.md [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+
+Run one instance of your producer program on one VM. The test should then be repeated for each message size, with a total of 5 test runs.
```
Comment (jmbeach) [inline DesignOfExperiments/Performance-MQ-MessageSize.md:117]:
```suggestion
Run one instance of your producer program on one VM. Configure the producer to always send the [tiny](./messages/tiny.txt) message.
```
I'd prefer to see all of the tests laid out explicitly for consistency.

**Review by TheENGR — APPROVED** [review]

**Pull request merged**